data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetState.java

```diff
@@ -138,7 +140,13 @@ public void handleRequest(RestApiRequest request) throws JsonProcessingException
         future.thenApply(
             maybeStateAndMetaData ->
                 maybeStateAndMetaData
-                    .map(stateAndMetaData -> AsyncApiResponse.respondOk(stateAndMetaData.getData()))
+                    .map(
+                        stateAndMetaData -> {
+                          if (stateAndMetaData.getMilestone() != SpecMilestone.PHASE0) {
+                            throw new BadRequestException("SpecMilestone not PHASE0");
+                          }
```

Comment:
this is the bit where we need to be careful, we can do this check, but we also need to check if our desired response is json before throwing this....

Reply:
Is it not always json once it gets to that point?